Clean up unused settings/experimental workpool flags

jaraics opened the pull request:
Usage of
PREFECT_EXPERIMENTAL_ENABLE_WORK_POOLS and PREFECT_EXPERIMENTAL_ENABLE_WORK_POOLS were removed in PrefectHQ#8362

There is no reference to them anywhere else, except in settings.py where they are declared. They are just clutter when you open the settings page in the self-hosted server.

<!-- 
Thanks for opening a pull request to Prefect! We've got a few requests to help us review contributions:

- Make sure that your title neatly summarizes the proposed changes.
- Provide a short overview of the change and the value it adds.
- Share an example to help us understand the change in user experience.
- Confirm that you've done common tasks so we can give a timely review.
- Review our contribution guidelines: https://docs.prefect.io/latest/contributing/overview/

Happy engineering!
-->

<!-- Include an overview here -->

### Example
<!-- 
Share an example of the change in action.

A code blurb is best. Changes to features should include an example that is executable by a new user.
If changing documentation, a link to a preview of the page is great.
 -->

### Checklist
<!-- These boxes may be checked after opening the pull request. -->

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] If this pull request adds new functionality, it includes unit tests that cover the changes
- [ ] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  <!-- If you do not have permission to add a label, a maintainer will add one for you -->

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed.

For new functions or classes in the Python SDK:

- [ ] This pull request includes helpful docstrings.
- [ ] If a new Python file was added, this pull request contains a stub page in the Python SDK docs and an entry in `mkdocs.yml` navigation.

bunchesofdonald commented on src/prefect/settings.py:
```diff
@@ -1391,16 +1391,6 @@ def default_cloud_ui_url(settings, value):
 Whether or not to enable flow run states on the flow run graph.
 """
 
-PREFECT_EXPERIMENTAL_ENABLE_WORK_POOLS = Setting(bool, default=True)
```
When we remove these experimental settings we also want to add them to the set of `REMOVED_EXPERIMENTAL_FLAGS` in this same file: https://github.com/PrefectHQ/prefect/blob/692642b545525a3617f83708b7bc976b23a3b68e/src/prefect/settings.py#L107-L114

The reason for this is removing the settings entirely causes the CLI to crash for anyone who has them in one or more of their profiles. Adding them to `REMOVED_EXPERIMENTAL_FLAGS` will make it so that the user is warned about it and they have time to take action.

jaraics replied:
Sure. Makes sense!

bunchesofdonald reviewed (changes requested):
Hi @jaraics, thanks for the contribution! I really appreciate you trying to help keep the codebase clean. I left one note about a change we should make, but otherwise this is great.

jaraics commented:
Hi @bunchesofdonald ! Thanks for taking the time and explaining the reasone. I appreciate that!
I've added them to REMOVED_EXPERIMENTAL_FLAGS. (Also added your explanation as comments to REMOVED_EXPERIMENTAL_FLAGS)

bunchesofdonald commented:
Hi @jaraics, it looks like this setting _is_ still used in places with the `experimental_parameter` decorator. Anywhere that decorator is used with the group of `work_pools` ultimately ties back to these flags. It's hard to find, I can see why you missed it.

Work pools have been in place for more than a year, I think we can safely remove the decorators if you're up for it.

bunchesofdonald commented on src/prefect/settings.py:
```diff
@@ -104,13 +104,18 @@
     "PREFECT_EXPERIMENTAL_WARN_EVENTS_CLIENT",
     "PREFECT_EXPERIMENTAL_ENABLE_FLOW_RUN_INFRA_OVERRIDES",
     "PREFECT_EXPERIMENTAL_WARN_FLOW_RUN_INFRA_OVERRIDES",
+    "PREFECT_EXPERIMENTAL_ENABLE_WORK_POOLS",
+    "PREFECT_EXPERIMENTAL_ENABLE_WORK_POOLS",
```
```suggestion
    "PREFECT_EXPERIMENTAL_ENABLE_WORK_POOLS",
```

jaraics commented:
> Hi @jaraics, it looks like this setting _is_ still used in places with the `experimental_parameter` decorator. Anywhere that decorator is used with the group of `work_pools` ultimately ties back to these flags. It's hard to find, I can see why you missed it.
> 
> Work pools have been in place for more than a year, I think we can safely remove the decorators if you're up for it.

Hey @bunchesofdonald I definatelly wish to take this home, I'm just having a hard time marrying a django project with prefect at this time (I wish there were tutorials/better support for it). Give me a couple days and I'll get back on this fix.

bunchesofdonald commented:
@jaraics looks like we're getting close, but there's a few test fails that seem related to this.

bunchesofdonald commented:
> @bunchesofdonald It seems the tests are for these flags. I think I should remove the tests. Is that acceptable?

I don't think we need to remove the tests here, the only 'true' failure I see is that the `work-queue ls` cli command is checking for the `work-pool` experiment. I think we can fix that up by removing this logic branch: https://github.com/PrefectHQ/prefect/blob/main/src/prefect/cli/work_queue.py#L399-L432

The others are our UI settings tests, you should just be able to remove `work_pools` from the sets in those tests.

Let me know if you run into issues!

jaraics commented:
Ok, I think I fixed the tests. (I can't be sure, they all time out on my PC, maybe I'm missing some setup. A devcontainer would be nice)

bunchesofdonald commented:
> @bunchesofdonald Ok, I think I fixed the tests. (I can't be sure, they all time out on my PC, maybe I'm missing some setup. A devcontainer would be nice)

The main failure is fixed. I took the liberty of fixing up the ui-settings tests here: https://github.com/PrefectHQ/prefect/pull/13144/commits/34ae3e9b4c3ee905c62fe4fd2ac9ac0920de5258

I'm sorry you're having issues with the tests, it's unusual that they would _all_ time out. The docker suite can timeout locally while building images, but the others tend to be fairly quick. There's not a whole lot of setup to do, I have a virtual env with Python 3.8, and to install the packages I use `pip install -e '.[dev]'`. One thing you might try is running with a single test runner with `pytest -n0` the default is to run one worker per CPU core, but that _could_ be overwhelming your PC.

A dev container _would_ be nice, would you mind writing up an issue?
